### PR TITLE
Add vrf best practices

### LINKED ISSUE #90: Add section to VRF for mapping requestId to answer
Give a demo of showing how to map your requestIds to the fulfilled randomness. ie something like:

```
mapping(bytes32 => uint256) requestIdToRandomNumber;

function fulfilRandomness(...)..{
requestIdToRandomNumber[requestId] =randomNumber;
}
```

Perhaps in a `bestPractices` section. We get this question in the discord fairly often.

### LINKED ISSUE #128: Add best practices page for VRF
New VRF Best Practices page, link to it from the main VRF page, and then extract the expand randomness notes too.
- “Getting more randomness (expanded randomness)”
- “Getting a random number within a range (using modulo)”
- “Having multiple requests for randomness in flight simultaneously (using maps and requestIds)”

### PR DESCRIPTION
**Pull request by dmitrydao**
fixes #128
fixes #90

### CODE REVIEW

File: docs/Using Randomness/vrf-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    2: 1459
+    3: 1459
+    4: "#dbe1f8"
+---
+
+Best are the practices for using Chainlink VRF.
+
+## 1. Getting a random number within a range
+
+If you need to generate a random number within a given range, you should use [modulo](https://docs.soliditylang.org/en/v0.5.3/types.html#modulo) to define the limits of your range. Below you can see how to get a random number between 1 and 50.
```
Comment (StephenFluin) [inline docs/Using Randomness/vrf-best-practices.md:23]: Shouldn't we link to a more recent version of Solidity?

File: docs/Using Randomness/vrf-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+```solidity
+uint256 public randomResult;
+
+function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
+    randomResult = (randomness % 50) + 1;
+}
+```
+
+## 2. Getting multiple random numbers
```
Comment (StephenFluin) [inline docs/Using Randomness/vrf-best-practices.md:33]: Remove header numbers if there isn't a sequence

File: docs/Using Randomness/vrf-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+```
+
+## 3. Having multiple VRF requests in flight
+
+If you want to have multiple VRF requests in flight, you should create a mapping between the request ID and the actual random number.
+
+```solidity
+mapping(bytes32 => uint256) requestIdToRandomNumber;
+
```
Comment (PatrickAlphaC) [inline docs/Using Randomness/vrf-best-practices.md:53]: You need to add this mapping in the initial request as well. Please change this to:
If you want to have multiple VRF requests in flight, you might want to create a mapping between the request ID and the requester.

Or, maybe you want to keep the ordering of when a request was made, in that case, you might want to use a mapping of `requestId` to index. 
```
mapping(bytes32 => address) public requestIdToAddress;
mapping(bytes32 => uint256) public requestIdToRequestNumber;
uint256 public counter;

function getRandomNumber(uint256 userProvidedSeed) public returns (bytes32 requestId) {
        bytes32 requestId =  requestRandomness(keyHash, fee, userProvidedSeed);
        requestIdToAddress[requestId] = msg.sender;
        requestIdToRequestNumber[requestId] = counter;
        counter += 1;
    }

function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
    address from = requestIdToAddress[requestId];
    uint256 requestNumber = requestIdToRandomNumberIndex[requestId];
}
```
Feel free to use whatever data structure you prefer.

**Review by PatrickAlphaC — CHANGES_REQUESTED** [review]

File: docs/Using Randomness/vrf-best-practices.md (hunk 3 above)
Comment (dmitrydao) [reply docs/Using Randomness/vrf-best-practices.md:57]: Added. Did we have any requests for keeping a track of multiple VRF results based on request ID?

Comment (PatrickAlphaC) [reply docs/Using Randomness/vrf-best-practices.md:72]: No, we could add that in too if you think it makes sense.

**Comment by PatrickAlphaC** [comment]
@dmitrydao where does this stand?

**Comment by PatrickAlphaC** [comment]
Can you update this to reflect the new API changes? `userProvidedSeed` is no longer needed. See https://github.com/smartcontractkit/documentation/pull/167

**Comment by dmitrydao** [comment]
> Can you update this to reflect the new API changes? `userProvidedSeed` is no longer needed. See #167

Oh, so you don't need a seed to get a random number, right?

**Comment by PatrickAlphaC** [comment]
> > Can you update this to reflect the new API changes? `userProvidedSeed` is no longer needed. See #167
> 
> Oh, so you don't need a seed to get a random number, right?

Yep! Brand new as of yesterday

**Comment by dmitrydao** [comment]
> > > Can you update this to reflect the new API changes? `userProvidedSeed` is no longer needed. See #167
> > 
> > 
> > Oh, so you don't need a seed to get a random number, right?
> 
> Yep! Brand new as of yesterday

Done

**Comment by PatrickAlphaC** [comment]
@dmitrydao looks good!

**Review by StephenFluin — APPROVED** [review]

**Pull request merged**